case-study.Rmd

```diff
@@ -0,0 +1,489 @@
+gsea_con2_sig <- gsea_con2[gsea_con2$`FDR q-val` <= 0.05, ]
+KEGG_GSEA <- unique(c(gsea_con1_sig$NAME, gsea_con2_sig$NAME))
+
+```
+
+Now let's compare the significant KEGG pathways in GSEA with the KEGG pathways
+associated with the active signatures.
+
+```{r}
+# GSEA automatically converts pathway names to upper case, so we also convert
+# the pathways identified by ADAGEpath.
+KEGG_ADAGE <- toupper(as.character(unique(pathway_association$geneset)))
+
+# make a venn diagram for pathway overlap
+grid.newpage()
+invisible(draw.pairwise.venn(length(KEGG_GSEA), length(KEGG_ADAGE),
+                             length(intersect(KEGG_GSEA, KEGG_ADAGE)),
+                             c("GSEA", "ADAGE"), fill = c("red", "blue")))
+```
+
+As we can see, GSEA and ADAGEpath agree on many pathways. Here we look into
+pathways significantly enriched in GSEA but not associated with chosen active
+signatures. We check whether they are assoicated with any ADAGE signatures and
+how differentially active the associated signatures are.
+
+```{r}
+# built a match table between normal case and upper case
+match_table <- data.frame(normal_case = names(KEGG_subset),
+                          upper_case = toupper(names(KEGG_subset)),
+                          stringsAsFactors = FALSE)
+
+# pathways enriched in GSEA but not associated with ADAGE signatures
+# in the top 10 layers of pareto fronts
+GSEA_only <- match_table$normal_case[match(setdiff(KEGG_GSEA, KEGG_ADAGE),
+                                           match_table$upper_case)]
+
+# check whether GSEA-only KEGG pathways are associated with any signatures
+# this step will take a while because it checks association between all signatures
+# and all pathways
+associated_sigs <- find_associated_signatures(input_genesets = GSEA_only,
+                                              model = model,
+                                              gene_set_list = KEGG_subset,
+                                              significance_cutoff = 0.05,
+                                              signature_limma_result = limma_result)
+associated_sigs <- unlist(associated_sigs)
+
+# see how these signatures perform in the activation test
+plot_volcano(limma_result, highlight_signatures = associated_sigs,
+             interactive = TRUE)
+```
+
+Here are pathways enriched in GSEA but not associated with any signatures:
+
+```{r}
+GSEA_only[is.na(associated_sigs)]
+```
+
+Here are pathways that were detected to be active in ADAGEpath but do not pass
+the significance cutoff in GSEA. Let's see how they perform in GSEA.
+
+```{r}
+ADAGE_active <- unique(combined_result$geneset[combined_result$adj.P.Val <= 0.05])
+ADAGE_only <- setdiff(toupper(ADAGE_active), KEGG_GSEA)
+ADAGE_only_GSEA <- rbind(gsea_con1[na.omit(match(ADAGE_only, gsea_con1$NAME)), ],
+                         gsea_con2[na.omit(match(ADAGE_only, gsea_con2$NAME)), ])
+knitr::kable(ADAGE_only_GSEA)
```

Comment:
The `NOM p-val` and `FDR q-val` columns are probably enough to interpret the result. I will add a note here.